packages/app/src/cli/services/dev/extension/bundler.ts

```diff
@@ -281,20 +312,30 @@ export async function setupFunctionWatcher({
         outputWarn(`Error while deploying updated extension draft: ${JSON.stringify(updateError, null, 2)}`, stdout)
       })
   })
+  listenForAbortOnWatcher(functionRebuildAndRedeployWatcher)
 
-  signal.addEventListener('abort', () => {
-    outputDebug(`Closing function file watching for extension with ID ${extension.devUUID}`, stdout)
-    functionWatcher
-      .close()
-      .then(() => {
-        outputDebug(`Function file watching closed for extension with ${extension.devUUID}`, stdout)
-      })
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      .catch((error: any) => {
-        outputDebug(
-          `Function file watching failed to close for extension with ${extension.devUUID}: ${error.message}`,
-          stderr,
+  if (redeployWatchPaths.length > 0) {
+    const functionRedeployWatcher = chokidar.watch(redeployWatchPaths).on('change', (path) => {
+      outputDebug(`File at path ${path} changed`, stdout)
+      if (buildController) {
+        // terminate any existing builds
+        buildController.abort()
```

Comment:
If these two share the same controller, is there a race condition here?

For example, if there's a file that trigger both watch conditions. I think the following events could happen:
- The first watch callback runs, which creates the controller
- The second watch callback runs, which aborts the controller
- `buildFunctionExtension` resolves, but `buildSignal` has been aborted so the function is not updated

To avoid this, I think we could use a different signal for the redeploy watcher.

Reply:
Thanks for pointing this out. I just fixed it. 

```
if (buildController) {
  // terminate any existing builds
  buildController.abort()
}
``` 
isn't required in the `functionRedeployWatcher` as it shouldn't be worried about rebuilding the function.